internal/librarian/stateandconfig.go

```diff
@@ -87,17 +91,74 @@ func parsePipelineConfig(contentLoader func() ([]byte, error)) (*config.Pipeline
 	return config, nil
 }
 
-func parseLibrarianState(contentLoader func() ([]byte, error)) (*config.LibrarianState, error) {
-	bytes, err := contentLoader()
+func parseLibrarianState(contentLoader func(file string) ([]byte, error), path, source string) (*config.LibrarianState, error) {
+	bytes, err := contentLoader(path)
 	if err != nil {
 		return nil, err
 	}
 	var s config.LibrarianState
 	if err := yaml.Unmarshal(bytes, &s); err != nil {
 		return nil, fmt.Errorf("unmarshaling librarian state: %w", err)
 	}
+	if err := populateServiceConfig(&s, contentLoader, source); err != nil {
```

Comment:
nit: can you rename to indicate it only populates when not already present in state.yaml?